### PR TITLE
Cache paths of fonts shipped with mpl relative to the mpl data path.

### PR DESCRIPTION
**Pull request by anntzer**
This lets the font cache stay valid across multiple virtualenvs (as long
as the list of fonts shipped by mpl does not change).

xref #10201.

As a side note note the previously buggy line `if fname.lower().find('DejaVuSans.ttf')>=0` which would always return False...

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by anntzer** [comment]
I think the gain of keeping the fontcache across venvs is nice enough to mark this as RC.

**Comment by tacaswell** [comment]
I think we need to add the matplotlib version number to the file incase we change the fonts we ship between versions again (to support multiple matplotlib versions across multiple venvs).

**Comment by anntzer** [comment]
Done as a separate commit.  Now also closes #4993 (until we decide to also change the format of the tex cache...).

**Comment by tacaswell** [comment]
```
" exited with 0.
53.00s$ if [[ $RUN_FLAKE8 == 1 ]]; then
  flake8 --statistics && echo "Flake8 passed without any issues!"
fi
./lib/matplotlib/font_manager.py:0:1: X100 Superfluous per-file-ignores for E225
1     X100 Superfluous per-file-ignores for E225

```

Cleaned up too much style!

**Comment by anntzer** [comment]
done

File: lib/matplotlib/font_manager.py [hunk 1]
```diff
@@ -828,15 +829,23 @@ def copy(self):
...
         elif isinstance(o, FontEntry):
-            return dict(o.__dict__, _class='FontEntry')
+            d = dict(o.__dict__, __class__='FontEntry')
+            try:
+                # Cache paths of fonts shipped with mpl relative to the mpl
+                # data path, which helps in the presence of venvs.
+                d["fname"] = str(
+                    Path(d["fname"]).relative_to(mpl.get_data_path()))
+            except ValueError:
+                pass
```
Comment (tacaswell) [inline lib/matplotlib/font_manager.py]: One concern is that are we sure every path we see here will be absolute?

Comment (anntzer) [reply lib/matplotlib/font_manager.py]: fontconfig returns absolute paths, and we call os.path.abspath in findSystemFonts.

**Pull request merged**